### ISSUE TITLE
Syntax error near unexpected token '(' from export of BASH_FUNC_module()

### ISSUE THREAD

**Issue by edickie** [pipitone/qbatch#61]
Hey guys, 

My jobs finish with the following error message:

```
/var/spool/torque/mom_priv/jobs/35134401-1.gpc-sched-ib0.SC: line 102: syntax error near unexpected token `('
/var/spool/torque/mom_priv/jobs/35134401-1.gpc-sched-ib0.SC: line 102: `export BASH_FUNC_module()="() {  if [ \"$$1\" = load ]; then'
```

Here is what the .array script looked like...
```sh
#!/bin/bash
#PBS -l nodes=1:ppn=8
#PBS -j oe
#PBS -o /bg01/homescinet/a/arisvoin/edickie/myscripts/abide/logs
#PBS -d /bg01/homescinet/a/arisvoin/edickie/myscripts/abide
#PBS -N Caltech
#PBS 
#PBS -t 1-7
#PBS -l walltime=2:00:00
#PBS 
#PBS 
# -- start copied env
export scratch="/scratch/a/arisvoin/edickie"
export MODULESHOME="/scinet/gpc/Modules6/Modules/3.2.9"
export MKLPATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/intel64"
export HCP_DATA="/scratch/a/arisvoin/edickie/ABIDE/hcp"
export SHELL="/bin/bash"
export HISTSIZE="1000"
export HCPPIPEDIR_tfMRIAnalysis="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//TaskfMRIAnalysis/scripts"
export MANPATH="/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/share/man:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/man/en_US:/scinet/gpc/tools/editors/nano/nano-2.2.4/share/man:/scinet/gpc/x11/6/man:/usr/local/share/man:/usr/share/man:/scinet/gpc/man:/scinet/gpc/Modules6/Modules/3.2.9/man:/scinet/gpc/man/:/usr/local/share/man:/usr/share/man:/scinet/gpc/man:/scinet/gpc/toolbin/x11/man:/scinet/gpc/man/:/scinet/gpc/Modules6/Modules/3.2.9/man:/scinet/gpc/man/:/scinet/gpc/share/man:/scinet/gpc/tools/gnuparallel/20140622/share/man"
export HCPPIPEDIR_Global="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//global/scripts"
export MKL_F95_MOD_ROOT="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/include/intel64"
export FSLOUTPUTTYPE="NIFTI_GZ"
export PYTHONPATH="/scinet/gpc/mpi/mpi4py/centos6/python2.7/openmpi-intel-1.4.x/lib/python2.7/site-packages/:/home/a/arisvoin/edickie/code/epitome/:/home/a/arisvoin/edickie/code/epitome/"
export FPATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/include"
export MIC_LIBRARY_PATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/lib/mic"
export GS_LIB="/scinet/gpc/x11/6/share/ghostscript/8.70/Resource/Init:/scinet/gpc/x11/6/share/ghostscript/8.70/Resource/Font:/scinet/gpc/x11/6/share/ghostscript/8.70/lib:/scinet/gpc/x11/6/fonts/default/ghostscript"
export PARALLEL="--workdir . --env PATH --env LD_LIBRARY_PATH --env LOADEDMODULES --env _LMFILES_ --env MODULE_VERSION --env MODULEPATH --env MODULEVERSION_STACK --env MODULESHOME --env OMP_DYNAMICS --env OMP_MAX_ACTIVE_LEVELS --env OMP_NESTED --env OMP_NUM_THREADS --env OMP_SCHEDULE --env OMP_STACKSIZE --env OMP_THREAD_LIMIT --env OMP_WAIT_POLICY"
export SCREENDIR="/scratch/a/arisvoin/edickie/.screen_run/"
export SCINET_X11_INC="/scinet/gpc/x11/6/include"
export HCPPIPEDIR_fMRIVol="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//fMRIVolume/scripts"
export VOLUME_CACHE_THRESHOLD="-1"
export MKLROOT="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/intel64"
export CXX="icpc"
export FSLGECUDAQ="cuda.q"
export HOSTNAME="gpc-f103n084-ib0"
export HCPPIPEDIR_PreFS="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//PreFreeSurfer/scripts"
export MNI_DATAPATH="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/data"
export MNI_DIR="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni"
export MAIL="/var/spool/mail/edickie"
export HCPPIPEDIR_PostFS="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//PostFreeSurfer/scripts"
export OS="Linux"
export FS_OVERRIDE="0"
export SCINET_MPI_LIB="/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/lib"
export GROUP="arisvoin"
export LESSOPEN="|/usr/bin/lesspipe.sh %s"
export FSLTCLSH="/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build/bin/fsltclsh"
export CPATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/include:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/include"
export USER="edickie"
export FSL_BIN="/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build/bin"
export MODULE_VERSION_STACK="3.2.9"
export FMRI_ANALYSIS_DIR="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/fsfast"
export SCINET_TBB_INC="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/include"
export HCPPIPEDIR_Templates="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//global/templates"
export SHLVL="1"
export SCINET_INTERACTIVE="1"
export EPITOME_DATA="/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain"
export MIC_LD_LIBRARY_PATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/lib/mic:/opt/intel/mic/coi/device-linux-release/lib:/opt/intel/mic/myo/lib:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/mic:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mpirt/lib/mic:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/mic"
export FREESURFER_HOME="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer"
export SCINET_X11_LIB="/scinet/gpc/x11/6/lib"
export CVSROOT="/usr/local/cvsroot"
export SCINET_NANO_BASE="/scinet/gpc/tools/editors/nano/nano-2.2.4"
export ENV="/home/a/arisvoin/edickie/.bashrc"
export BASH_ENV="/home/a/arisvoin/edickie/.bashrc"
export _="/home/a/arisvoin/edickie/code/qbatch/bin/qbatch"
export MODULEPATH="/scinet/gpc/Modules6/Modules/versions:/scinet/gpc/Modules6/Modules/$$MODULE_VERSION/modulefiles:/scinet/gpc/Modules6/Modules/modulefiles:/home/t/tpaus/tpaus/privatemodules:/home/a/arisvoin/edickie/quarantine/modules:/home/a/arisvoin/edickie/quarantine//modules"
export PERL5LIB="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/lib/perl5/5.8.5:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/lib/perl5/5.8.5:/home/t/tpaus/tpaus/opt/minc/perl/lib/perl5/site_perl/5.8.8:"
export SCINET_FONTPATH="/scinet/gpc/x11/6/fonts/largerset"
export SSH_CONNECTION="172.20.82.14 38764 172.20.250.251 22"
export FSLMULTIFILEQUIT="TRUE"
export FIX_VERTEX_AREA=""
export HCPPIPEDIR_FS="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//FreeSurfer/scripts"
export PBS_DEFAULT="gpc-sched-ib0"
export HCPPIPEDIR_fMRISurf="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//fMRISurface/scripts"
export SUBJECTS_DIR="/scratch/a/arisvoin/edickie/ABIDE/FSout"
export _LMFILES_="/scinet/gpc/Modules6/Modules/modulefiles/extras/64_6.4:/scinet/gpc/Modules6/Modules/modulefiles/Xlibraries/X11-64:/scinet/gpc/Modules6/Modules/modulefiles/ImageMagick/6.6.7:/scinet/gpc/Modules6/Modules/modulefiles/nano/2.2.4:/scinet/gpc/Modules6/Modules/modulefiles/intel/14.0.1:/scinet/gpc/Modules6/Modules/modulefiles/openmpi/1.4.4-intel-v12.1:/scinet/gpc/Modules6/Modules/modulefiles/gnu-parallel/20140622:/home/a/arisvoin/edickie/quarantine/modules/edickie_quarantine:/home/a/arisvoin/edickie/quarantine/modules/FSL/5.0.9:/home/a/arisvoin/edickie/quarantine/modules/AFNI/2015.12.15:/home/a/arisvoin/edickie/quarantine/modules/FIX/1.06:/home/a/arisvoin/edickie/quarantine/modules/connectome-workbench/1.1.1:/home/a/arisvoin/edickie/quarantine/modules/hcp-pipelines/3.13.1:/home/a/arisvoin/edickie/quarantine/modules/python/2.7.11-Anaconda2-2.5.0"
export HCPPIPEDIR_dMRITract="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//DiffusionTractography/scripts"
export MKL_HOME="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl"
export MODULE_VERSION="3.2.9"
export FSLDISPLAY="/usr/bin/display"
export INTEL_LICENSE_FILE="/scinet/gpc/intel/licenses"
export HCPPIPEDIR_dMRI="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//DiffusionPreprocessing/scripts"
export SCINET_VIZ="0"
export FC="ifort"
export HOME="/home/a/arisvoin/edickie"
export HCPPIPEDIR_Config="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//global/config"
export LD_LIBRARY_PATH="/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/lib/openmpi:/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/lib:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/lib/intel64:/scinet/gpc/x11/6/lib:/scinet/gpc/toolbin/x11/lib:/scinet/gpc/lib64_6.4:/scinet/gpc/lib64_6.4/mysql:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/intel64:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/debugger/lib/intel64:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/intel64:/opt/intel/mic/coi/host-linux-release/lib:/opt/intel/mic/myo/lib"
export LANG="C"
export LIBRARY_PATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/lib/intel64:/scinet/gpc/lib64_6.4:/scinet/gpc/lib64_6.4/mysql:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/intel64:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/debugger/lib/intel64:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/intel64"
export FSF_OUTPUT_FORMAT="nii.gz"
export SCINET_OTPO_BASE="/scinet/gpc/mpi/openmpi/otpo/1.0.0/"
export NLSPATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/intel64/locale/en_US"
export G_BROKEN_FILENAMES="1"
export SCINET_EXTRAS_INC="/scinet/gpc/include6"
export SCINET_JOBID=""
export SCINET_INTEL_BASE="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/"
export MSMBin="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//MSMBinaries"
export CC="icc"
export SINK_LD_LIBRARY_PATH="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/mic:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/lib/mic:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mpirt/lib/mic"
export SCINET_TBB_LIB="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb/lib/intel64"
export TBBROOT="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/tbb"
export BASH_FUNC_module()="() {  if [ \"$$1\" = load ]; then
 shift;
 for each_module in $$*;
 do
 eval `/scinet/gpc/Modules6/Modules/3.2.9/bin/modulecmd bash load $$each_module`;
 done;
 unset each_module;
 else
 if [ \"$$1\" = advice ]; then
 shift;
 /scinet/gpc/bin6/moduleadvice $$*;
 else
 if [ \"$$1\" = find ]; then
 shift;
 /scinet/gpc/bin6/modulefind $$*;
 else
 eval `/scinet/gpc/Modules6/Modules/3.2.9/bin/modulecmd bash $$*`;
 fi;
 fi;
 fi
}"
export SCINETVIZDONE="1"
export F77="ifort"
export SSH_TTY="/dev/pts/19"
export QUARANTINE_ENV="scinet"
export SCINET_X11_BIN="/scinet/gpc/x11/6/bin"
export GDB_CROSS="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/debugger/gdb/intel64_mic/py26/bin/gdb-mic"
export CVS_RSH="ssh"
export SCINET_INTEL_INC="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/include"
export LOCAL_DIR="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/local"
export CARET7DIR="/home/a/arisvoin/edickie/quarantine//connectome-workbench/1.1.1/build/bin_rh_linux64"
export FSLCONVERT="/usr/bin/convert"
export FSLDIR="/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build"
export MKL="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/mkl/"
export SCINET_EXTRAS_LIB="/scinet/gpc/lib64_6.4"
export SSH_CLIENT="172.20.82.14 38764 22"
export LOGNAME="edickie"
export PATH="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/fsfast/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/tktools:/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/bin:/home/a/arisvoin/edickie/quarantine//python/2.7.11-Anaconda2-2.5.0/build/bin:/home/a/arisvoin/edickie/quarantine//connectome-workbench/1.1.1/build/bin_rh_linux64:/home/a/arisvoin/edickie/quarantine//FIX/1.06/build:/home/a/arisvoin/edickie/quarantine//AFNI/2015.12.10/build:/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/fsfast/bin:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/tktools:/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/bin:/scinet/gpc/mpi/openmpi/otpo/1.0.0//bin:/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/bin:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/bin/intel64:/scinet/gpc/graphics/ffmpeg-0.8.2-centos6/bin:/scinet/gpc/graphics/ImageMagick-6.6.7-7-centos6/bin:/scinet/gpc/x11/6/bin:/usr/local/bin:/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/scinet/gpc/toolbin:/scinet/gpc/toolbin/vnc:/scinet/gpc/toolbin/x11/bin:/usr/lpp/mmfs/bin:/opt/torque/bin:/opt/torque/sbin:/scinet/gpc/bin6:/scinet/gpc/x11/6/bin/old:/scinet/gpc/tools/editors/nano/nano-2.2.4/bin:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/bin/intel64_mic:/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/debugger/gdb/intel64_mic/py26/bin:/opt/intel/mic/bin:/scinet/gpc/tools/gnuparallel/20140622/bin:/home/a/arisvoin/edickie/code/epitome/bin:/home/a/arisvoin/edickie/code/epitome/bin"
export SCINET_INTEL_LIB="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/compiler/lib/intel64"
export TERM="xterm"
export FSL_DIR="/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build"
export MNI_PERL5LIB="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/lib/perl5/5.8.5"
export EPITOME_CLONE="/home/a/arisvoin/edickie/epi-clone"
export FSLWISH="/home/a/arisvoin/edickie/quarantine//FSL/5.0.9/build/bin/fslwish"
export HCPPIPEDIR_Bin="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//global/binaries"
export SCINET_MPI_BASE="/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1"
export MINC_BIN_DIR="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/bin"
export SCINET_MPI_INC="/scinet/gpc/mpi/openmpi/1.4.4-intel-v12.1/include"
export FSFAST_HOME="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/fsfast"
export PROJECT="/project/a/arisvoin/edickie"
export SCINET_USE_LMOD="false"
export MINC_LIB_DIR="/scratch/project/t/tpaus/tpaus/FREESURFER/freesurfer/mni/lib"
export edickie_QUARANTINE_PATH="/home/a/arisvoin/edickie/quarantine/"
export HCPPIPEDIR="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build/"
export HCPPIPEDIR_tfMRI="/home/a/arisvoin/edickie/quarantine//hcp-pipelines/3.13.1/build//tfMRI/scripts"
export GDBSERVER_MIC="/scinet/gpc/intel/ics/composer_xe_2013_sp1.1.106/debugger/gdb/target/mic/bin/gdbserver"
export SCRATCH="/scratch/a/arisvoin/edickie"
export OLDPWD="/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Yale"
export LOADEDMODULES="extras/64_6.4:Xlibraries/X11-64:ImageMagick/6.6.7:nano/2.2.4:intel/14.0.1:openmpi/1.4.4-intel-v12.1:gnu-parallel/20140622:edickie_quarantine:FSL/5.0.9:AFNI/2015.12.15:FIX/1.06:connectome-workbench/1.1.1:hcp-pipelines/3.13.1:python/2.7.11-Anaconda2-2.5.0"
export HISTCONTROL="ignoredups"
export SCINET_LOGIN="0"
# -- end copied env

ARRAY_IND=PBS_ARRAYID
CHUNK_SIZE=4
CORES=8
sed -n "$(( (${ARRAY_IND} - 1) * ${CHUNK_SIZE} + 1 )),+$(( ${CHUNK_SIZE} - 1 ))p" << EOF | parallel -j${CORES}
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51456/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51458/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51461/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51462/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51463/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51466/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51470/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51471/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51472/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51475/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51476/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51477/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51478/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51479/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51480/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51481/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51482/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51483/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51484/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51485/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51487/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51489/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51491/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51492/cmd_160319_104903_20160319.sh
/scratch/a/arisvoin/edickie/ABIDE/epi-fixtrain/Caltech/Caltech_51493/cmd_160319_104903_20160319.sh
```

**Comment by gdevenyi** [comment]
Jobs and the module system are a recurring issue

Questions:
1. Did the jobs finish successfully?
2. What is the contents of your .bashrc and .bash_profile?

**Comment by edickie** [comment]
My .bashrc and bash_profiles are below...
There's not mush in it but i do load a lot of modules right before running
qbatch

##################################################
if [ -f /etc/bashrc ]; then
       . /etc/bashrc
fi

# commands which work for both GPC and TCS can go here

#setting $scratch
export scratch=$SCRATCH

# setting alias to the camh scc
alias scc='edickie@192.197.205.74'

#adding this to PATH
#export PATH=$PATH:/home/t/tpaus/tpaus/opt/pipelines
#export PATH=/home/t/tpaus/tpaus/opt/lib/X11:$PATH

#other enviroment stuff
export
PERL5LIB=/home/t/tpaus/tpaus/opt/minc/perl/lib/perl5/site_perl/5.8.8:$PeRL5LIB
export VOLUME_CACHE_THRESHOLD=-1
export CVSROOT=/usr/local/cvsroot

#export LD_LIBRARY_PATH=/home/t/tpaus/tpaus/opt/minc_prereq/lib

HOST=$(uname)

if [ "${HOST}" == "AIX" ]; then
    # do things for the TCS machine
    # user environment for all shells goes here
    # replace colon with your own commands
    :
else
    # do things for the GPC machine


    module load Xlibraries extras ImageMagick nano
    module unload intel openmpi
    module load intel/14.0.1 openmpi
    module use --append /home/t/tpaus/tpaus/privatemodules

    ### for fsl/fix to work
    ##module load R/3.0.1 hdf5/187-v18-serial-intel octave/3.8.1
    #module purge
    #module load extras intel/14.0.1 openmpi/intel/1.6.4 R/3.1.1
hdf5/187-v18-serial-intel octave/3.8.1

    # user environment for all shells goes here
    # replace colon with your own commands
    :
fi

####################################################
if [ -f /etc/profile ]; then
       . /etc/profile
fi

# commands which work for both GPC and TCS can go here

alias passwd='echo "Please use the SciNet portal to change password:
https://portal.scinet.utoronto.ca/change_password"'

HOST=$(uname)

if [ "${HOST}" == "AIX" ]
then
    # do things for the TCS machine
    alias llq1='/xcat/tools/tcs-scripts/LL/jobState.sh'
    alias llstat='/xcat/tools/tcs-scripts/LL/jobSummary.sh'

    if [ "${TERM}" = "xterm-color" ]; then
        export TERM=xterm
    fi

    # user environment for login shells goes here
    # replace colon with your own commands
    :
else
    # do things for the GPC machine
    # user environment for login shells goes here
    # replace colon with your own commands
    :

fi

PS1="\h-\$ "


if [ -f ~/.bashrc ]; then
       . ~/.bashrc
fi

## FSL Setup
#FSLDIR=/home/j/jlerch/edickie/fsl5/fsl
#PATH=${FSLDIR}/bin:${PATH}
#export FSLDIR PATH
#. ${FSLDIR}/etc/fslconf/fsl.sh

**Comment by gdevenyi** [comment]
Did the jobs work?

**Comment by gdevenyi** [comment]
You are loading modules in your .bashrc/.bash_profile

From what I recall, those are the origin of the modules errors you posted. Please remove those and try again.

**Comment by pipitone** [comment]
We could add `BASH_FUNC_module()` to our list of environment vars to exclude from export... Or perhaps just exclude all vars matching `[()]` :-)

**Comment by gdevenyi** [comment]
We should test of excluding that breaks module loading in the job script preamble

**Comment by pipitone** [comment]
FYI, just tested this on scinet and I am also seeing this same error. I am investigating.

**Comment by pipitone** [comment]
Okay, if I remove the export, the jobs run fine. Question is, why *aren't* you seeing this error on scinet? 

If I run a job script modified to not have the module() export and to contain a module load statement, then I do get an error when trying to load modules as expected: 

```
$ env -i PBS_ARRAYID=1 .scripts/qbatch.test.array                                                                                     
.scripts/qbatch.test.array: line 94: module: command not found
```

Oh `environment-modules`... what a mess.

**Comment by pipitone** [comment]
Okay, I've submitted a fix to ignore the `BASH_FUNC_module()` environment variable. @gdevenyi, what do you think of this?

**Comment by gdevenyi** [comment]
Does this allow modules to be loaded in the job? Or is that now broken?

**Comment by pipitone** [comment]
> Does this allow modules to be loaded in the job? Or is that now broken?

It's broken on scinet now (unless we switch back to using bash -l). I
don't yet understand it, but bash on Ubuntu (and the same version of
Bash on Centos here at CAMH) doesn't export the BASH_FUNC_modules()
variable so isn't broken by this change.

**Comment by pipitone** [comment]
I'd still like to know more about how you have things configured on
Scinet such that you aren't seeing this error. What does your .bashrc
look like?

**Comment by gdevenyi** [comment]
I actually didn't say anywhere than I don't see those errors... I haven't tested the latest version with module saving directly on scinet.

Furthermore, on SGE here at the CIC, I get that error with the latest qbatch.

**Comment by pipitone** [comment]
> I haven't tested the latest version with module saving directly on scinet.

What do you mean by "module saving"? 

> Furthermore, on SGE here at the CIC, I get that error with the latest qbatch.

Latest, as in, including the fix to ignore "BASH_FUNC_module()"?

**Comment by gdevenyi** [comment]
** environment saving intrinsic to qbatch, rather than using the scheduler system to handle it.

Latest as in immediately prior to the "fix"

**Comment by gdevenyi** [comment]
Fun fact, this isn't an SGE issue, this in an importing bash environment issue.

Running python tests for an (unrelated change):
```
======================================================================
FAIL: test_qbatch.test_run_qbatch_local_piped_commands
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/data/chamal/projects/gabriel/src/qbatch/test/test_qbatch.py", line 106, in test_run_qbatch_local_piped_commands
    assert out == expected, out
AssertionError: cmdTrace.c(713):ERROR:104: '5857*' is an unrecognized subcommand
cmdModule.c(411):ERROR:104: '5857*' is an unrecognized subcommand
/data/chamal/projects/gabriel/src/qbatch/tmp1Ic79Y/STDIN.0: line 34: export: `BASH_FUNC_module%%=() {  eval
}': not a valid identifier
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
hello
```

This is of course, without the "fix"

**Comment by gdevenyi** [comment]
See PR #67 for a broader fix to ignore function exports.

**Comment by gdevenyi** [comment]
Fixed by #68
